beacon-chain/powchain/service.go

```diff
@@ -128,7 +130,8 @@ type Service struct {
 	ctx                     context.Context
 	cancel                  context.CancelFunc
 	headTicker              *time.Ticker
-	currHttpEndpoint        string
+	httpEndpoints           []httputils.Endpoint
+	currHttpEndpoint        *httputils.Endpoint
```

Comment:
can we not make this a pointer ? If we are making this a pointer then the above slice probably should be a pointer too